src/WebJobs.Script/Binding/Http/HttpTriggerAttribute.cs

```diff
@@ -13,5 +13,20 @@ public HttpTriggerAttribute()
         }
 
         public string RouteTemplate { get; set; }
+
+        // Handle "Route" vs. "RouteTemplate" naming mismatch. 
+        // We could get rid of these if the names matched. 
+        private class HttpTriggerAttributeMetadata : AttributeMetadata
```

Comment:
Ideally, Function.Json and the attributes would be 100% aligned and then we can seamlessly convert between them without any additional glue code. However, we have various diverges (slightly different property names, etc). In those cases, we can now create a "hint" class like this one to aide us in transition.  It's still a net reduction in code; and new extensions will not require this because they'll be consistent from the start.